### PR TITLE
fix: prevent double listing of elementStates when creating TopologyCase

### LINKED ISSUE #62: fix: prevent double listing of elementStates when creating TopologyCase
**Problem:** The `ElementState` for a object, which is out of service itself and is also connected to a node that is out of service too, will be created twice. This is not necessary and also not unique.

**Proposed solution:** in function `create_node_power_on_states` filter/neglect elements which are out of service itself

**Additional:** the object type of externalGrid should be considered when creating elementStates in function `create_topology_case()`

### PR DESCRIPTION
**Pull request by SebastianDD**
Fixes #62

### CODE REVIEW

**Comment by SebastianDD** [comment]
throws error:
``` File ~\Repos\powerfactory-utils\powerfactory_utils\exporter\exporter.py:1725, in PowerfactoryExporter.create_topology_case(self, meta, data)
   1715 power_on_states = self.pfi.list_from_sequences(
   1716     switch_states,
   1717     coupler_states,
   (...)
   1721     element_power_on_states,
   1722 )
   1723 power_on_states = self.merge_power_on_states(power_on_states)
-> 1725 return TopologyCase(meta=meta, elements=power_on_states)
...
File d:\Users\s9842011\Repos\powerfactory-utils\.venv\lib\site-packages\pydantic\main.py:342, in pydantic.main.BaseModel.__init__()

ValidationError: 1 validation error for Case
elements
  unhashable type: 'list' (type=type_error)
```

may conversion to tuple helps?

**Comment by SebastianDD** [comment]
also regarding issue #64 : I think the attribute `active` will be obsolete

**Comment by sasanjac** [comment]
> unhashable type: 'list' (type=type_error)

I would rather keep it because it is the single element that indicates wether a device is active or not regardless of reason. If we remove it, you have to check if `disabled is True` or if `len(open_switches) >= 0`

**Comment by SebastianDD** [comment]
> > unhashable type: 'list' (type=type_error)
> 
> I would rather keep it because it is the single element that indicates wether a device is active or not regardless of reason. If we remove it, you have to check if `disabled is True` or if `len(open_switches) >= 0`

sure, but on the other hand side, only elements which are either `disabled `and/or have `open_switches `should be listed in TopologyCase anyway, shouldn't it?

**Comment by sasanjac** [comment]
True, but then maybe we should rename `TopologyState` to `InactiveElements`

File: README.md [hunk 1]
```diff
@@ -17,9 +18,12 @@ Therefore, the Python-PowerFactory-API, provided by the company, is utilized.
 The following functionalities are provided:
 
 * export of calculation relevant grid data from a PowerFactory project into three common readable JSON files utilizing predefined [schemas](./powerfactory_utils/schema):
-  * grid topology
-  * topology case
+  * grid topology:
+    * base topology containing all elements of the exported grid
+  * topology case;
+    * information about disabled elements to represent a specific operational cases based on the base topology
```
Comment (SebastianDD) [inline README.md:24]:
```suggestion
    * information about disabled elements to represent a specific operational case based on the base topology
```

**Review by sasanjac — APPROVED** [review]

**Pull request merged**